SmartDeviceLink/private/SDLTCPTransport.m

```diff
@@ -271,7 +271,9 @@ - (void)sdl_readFromStream {
     }
 
     NSData *data = [NSData dataWithBytesNoCopy:buffer length:(NSUInteger)readBytes freeWhenDone:YES];
-    [self.delegate onDataReceived:data];
+    dispatch_after(DISPATCH_TIME_NOW, SDLGlobals.sharedGlobals.sdlProcessingQueue, ^{
+        [self.delegate onDataReceived:data];
+    });
```

Comment:
`dispatch_async` or `dispatch_sync` is probably better than `dispatch_after`. Also, add a comment about the cases why this is necessary.